app/helpers/fact_values_helper.rb

```diff
@@ -45,13 +45,15 @@ def show_full_fact_value(fact_value)
     end.html_safe
   end
 
-  def fact_origin_icon(origin)
+  # Return fact icon image tag. Stub is used in some cases because
+  # of legal requirements. Make sure to get legal advice prior
+  # putting any logos into our git repository.
+  def fact_origin_icon(origin, opts = {})
     return origin if origin == 'N/A'
-    title = origin
-    if origin.include? "/"
-      title = origin.split("/")[1]
-    end
-    image_tag(origin + ".png", :title => title)
+    origin = origin.split("/")[1] if origin.include?("/")
+    size = opts[:size] ||= '16x16'
+    filename = (origin =~ /Puppet/) ? "stub/goldenrod-p" : origin
```

Comment:
Note that we define the icon for plugins here.
https://github.com/theforeman/foreman_ansible/blob/f5b602a76c925f147f3ff74194b496a38c1cdc69/app/models/foreman_ansible/fact_name.rb#L8-L10

Puppet specific code should go here:
https://github.com/theforeman/foreman/blob/develop/app/models/fact_names/puppet_fact_name.rb

Reply:
Hmmm I cannot use origin, it's defined (and documented) as origin name, e.g. Puppet or Ansible. Ansible plugin uses that in a weird way, let me go step back and define this properly - I am going to fix `origin` to be used properly and introduce new method `icon_path` to return what's really needed. Plugins can redefine, sane default will be provided.

Reply:
I think the point here is, you replaced the logic how we determine icons for facts from various soruces. Ansible, Chef, Salt and Katello relies on this. Unless you implement `icon_path` method for their parsers, they will end up with `icons16x16/stub/black-x.png`.

Reply:
which is now resolved I think by sending PRs to other repos